interface/src/app/gene/network/network.js

```diff
@@ -175,57 +193,52 @@ angular.module('adage.gene.network', [
         }
 
         /**
-         * Callback function to show edge tips when mouse is over an edge.
+         * Callback function to show gene tips when a gene is clicked.
+         * @param {gene_data} data;
+         * @return {void}.
+         */
+        function showGeneTip(data) {
+          edgeTip.hide(); // Hide edge-tip window (if any) first.
+          geneTip.show(data);
+        }
+
+        /**
+         * Callback function to show edge tips when an edge is clicked.
          * @param {edge_data} data;
          * @return {void}.
          */
         function showEdgeTip(data) {
+          geneTip.hide(); // Hide gene-tip window (if any) first.
           var rawWeight = data.weight + rawMinWeight;
-          var str = 'Edge weight: ' + rawWeight.toFixed(3);
+          var htmlText = 'Edge weight: ' + rawWeight.toFixed(3);
           var heavyGenes = [data.gene1.id, data.gene2.id].join(',');
           var target = d3.event.target;
           NodeService.get(
             {'heavy_genes': heavyGenes, 'limit': 0},
             function success(response) {
-              var n = response.objects.length;
-              str += '<br />' + n + (n > 1 ? ' nodes are ' : ' node is ');
-              str += 'related to both genes' + (n > 0 ? ':' : '.');
-              for (var i = 0; i < n; ++i) {
-                str += '<br />* ' + response.objects[i].name + '</li>';
+              var i = 0, n = response.objects.length;
+              var anchorTag;
+              htmlText += '<br />' + n + (n > 1 ? ' nodes are ' : ' node is ');
+              htmlText += 'related to both genes' + (n > 0 ? ':' : '.');
+              for (; i < n; ++i) {
+                anchorTag = '<a href="#/node/' + response.objects[i].id + '">';
+                htmlText += '<br />* ' + anchorTag + response.objects[i].name;
+                htmlText += '</a>';
               }
-
-              edgeTip.html(str);
+              edgeTip.html(htmlText);
               edgeTip.show(data, target);
             },
-            function error(err) {
-              $log.error(
-                'Failed to get node info for gene edge: ' + err.statusText);
-              str += 'Can\'t get node information, please try again later.';
-              edgeTip.html(str);
+            function error(response) {
+              var message = 'Failed to get node info for gene edge: ' +
+                  response.statusCode + ' ' + response.statusText;
+              $log.error(message);
```

Comment:
:+1:  Nice logging of the error, and also handling the case when there is an error from the backend and how to present it to the user.